src/appengine/handlers/upload_testcase.py

```diff
@@ -177,18 +179,19 @@ def is_greybox_fuzzer(name):
   return [f for f in fuzzers if not is_greybox_fuzzer(f)]
 
 
+@memoize.wrap(memoize.Memcache(MEMCACHE_TTL_IN_SECONDS))
 def find_fuzz_target(engine, target_name, job_name):
-  """Find a fuzz target given the engine, target name (which may or may not be
-  prefixed with project), and job."""
+  """Return fuzz target values given the engine, target name (which may or may
+  not be prefixed with project), and job."""
   project_name = data_handler.get_project_name(job_name)
   candidate_name = data_types.fuzz_target_fully_qualified_name(
       engine, project_name, target_name)
 
   target = data_handler.get_fuzz_target(candidate_name)
-  if target:
-    return target.fully_qualified_name(), target.binary
+  if not target:
+    raise helpers.EarlyExitException('Fuzz target does not exist.', 400)
```

Comment:
This avoids memcache to save (None, None) tuple and we raise exception in caller anyway, so just moved it here.